common/archiver/archivalMetadata.go

```diff
@@ -177,7 +183,7 @@ func (a *archivalConfig) ReadEnabled() bool {
 	if !a.ClusterConfiguredForArchival() {
 		return false
 	}
-	return a.enableRead()
+	return a.staticEnableRead && a.dynamicEnableRead()
```

Comment:
This configuration might be surprising as some folks might expect either `||` or `if dynamicConfig != nil return dynamicConfig else staticConfig`

Reply:
I *dont* have a strong opinion on what it should be as I don't know enough about how this and similar config values are used in production, but I think maybe we should call out this both in the head of the diff and maybe in comments?

Reply:
This is to keep the behavior the same as GetClusterStatus function